public/js/factories/achievementsResourceFactory.js

```diff
@@ -0,0 +1,51 @@
+netstix.factory('AchievementsResource', ['$http', '$q', function($http, $q) {
+  var achievementsResource = {};
+
+  achievementsResource.postAchievements = function(title, criteria, points, challengeRepo, badgeLink) {
```

Comment:
Awesome stuff. Did $resource not work? I think it also just returns promises which would make this easier